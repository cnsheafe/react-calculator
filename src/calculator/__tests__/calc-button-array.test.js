import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import CalcButtonArray from '../calc-button-array'

configure({ adapter: new Adapter() })

describe('<CalcButton />', () => {
  const symbols = [
    { text: '+', onClick: jest.fn()},
    { text: '-', onClick: jest.fn()},
    { text: '=', onClick: jest.fn()}
  ]

  it('should match the snapshot', () => {
    const wrapper = shallow(<CalcButtonArray symbols={symbols} uniqueName="calc-1" />)
    expect(wrapper).toMatchSnapshot()
  })
})