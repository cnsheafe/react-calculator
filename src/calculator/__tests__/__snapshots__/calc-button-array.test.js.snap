// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalcButton /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalcButtonArray
    defaultHandler={[Function]}
    keyPrefix="calc-1"
    symbols={
        Array [
            Object {
              "onClick": [Function],
              "text": "+",
            },
            Object {
              "onClick": [Function],
              "text": "-",
            },
            Object {
              "onClick": [Function],
              "text": "=",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          +
</button>,
        <button
          onClick={[Function]}
>
          -
</button>,
        <button
          onClick={[Function]}
>
          =
</button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "calc-1-+",
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "calc-1--",
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "calc-1-=",
        "nodeType": "host",
        "props": Object {
          "children": "=",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "=",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            +
</button>,
          <button
            onClick={[Function]}
>
            -
</button>,
          <button
            onClick={[Function]}
>
            =
</button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "calc-1-+",
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "calc-1--",
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "calc-1-=",
          "nodeType": "host",
          "props": Object {
            "children": "=",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "=",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
